webapp/IronCalc/src/components/RightDrawer/RightDrawer.tsx

```diff
@@ -0,0 +1,186 @@
+import type { DefinedName, WorksheetProperties } from "@ironcalc/wasm";
+import { styled } from "@mui/material/styles";
+import type { MouseEvent as ReactMouseEvent, ReactNode } from "react";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { theme } from "../../theme";
+import { TOOLBAR_HEIGHT } from "../constants";
+import NamedRanges from "./NamedRanges/NamedRanges";
+
+const DEFAULT_DRAWER_WIDTH = 360;
+const MIN_DRAWER_WIDTH = 300;
+const MAX_DRAWER_WIDTH = 500;
+
+interface RightDrawerProps {
+  isOpen: boolean;
+  onClose: () => void;
+  width?: number;
+  onWidthChange?: (width: number) => void;
+  children?: ReactNode;
+  showCloseButton?: boolean;
+  backgroundColor?: string;
+  title?: string;
+  definedNameList?: DefinedName[];
+  worksheets?: WorksheetProperties[];
+  updateDefinedName?: (
+    name: string,
+    scope: number | undefined,
+    newName: string,
+    newScope: number | undefined,
+    newFormula: string,
+  ) => void;
+  newDefinedName?: (
+    name: string,
+    scope: number | undefined,
+    formula: string,
+  ) => void;
+  deleteDefinedName?: (name: string, scope: number | undefined) => void;
+  selectedArea?: () => string;
+}
+
+const RightDrawer = ({
+  isOpen,
+  onClose,
+  width = DEFAULT_DRAWER_WIDTH,
+  onWidthChange,
+  children,
+  showCloseButton = true,
+  title = "Named Ranges",
+  definedNameList,
+  worksheets,
+  updateDefinedName,
+  newDefinedName,
+  deleteDefinedName,
+  selectedArea,
+}: RightDrawerProps) => {
+  const [drawerWidth, setDrawerWidth] = useState(width);
+  const [isResizing, setIsResizing] = useState(false);
+  const resizeHandleRef = useRef<HTMLDivElement>(null);
+
+  // Update local width when prop changes
+  useEffect(() => {
+    setDrawerWidth(width);
+  }, [width]);
+
+  const handleMouseDown = useCallback((e: ReactMouseEvent) => {
+    e.preventDefault();
+    setIsResizing(true);
+  }, []);
+
+  useEffect(() => {
+    if (!isResizing) return;
+
+    // Prevent text selection during resize
+    document.body.style.userSelect = "none";
+    document.body.style.cursor = "col-resize";
+
+    const handleMouseMove = (e: MouseEvent) => {
+      const newWidth = window.innerWidth - e.clientX;
+      const clampedWidth = Math.max(
+        MIN_DRAWER_WIDTH,
+        Math.min(MAX_DRAWER_WIDTH, newWidth),
+      );
+      setDrawerWidth(clampedWidth);
+      onWidthChange?.(clampedWidth);
+    };
+
+    const handleMouseUp = () => {
+      setIsResizing(false);
+      document.body.style.userSelect = "";
+      document.body.style.cursor = "";
+    };
+
+    document.addEventListener("mousemove", handleMouseMove);
+    document.addEventListener("mouseup", handleMouseUp);
```

Comment:
This might be ok, but it would be much better if we can avoid attaching events to the document.

you can also leave that to me.